public/app/features/variables/pickers/OptionsPicker/OptionsPicker.tsx

```diff
@@ -16,125 +16,125 @@ import { formatVariableLabel } from '../../shared/formatVariable';
 
-  onToggleOption = (option: VariableOption, clearOthers: boolean) => {
-    const toggleFunc = this.props.variable.multi ? this.onToggleMultiValueVariable : this.onToggleSingleValueVariable;
-    toggleFunc(option, clearOthers);
-  };
+  class OptionsPickerUnconnected extends PureComponent<Props> {
+    onShowOptions = () => this.props.showOptions(this.props.variable);
+    onHideOptions = () => this.props.commitChangesToVariable(this.props.onVariableChange);
 
-  onToggleSingleValueVariable = (option: VariableOption, clearOthers: boolean) => {
-    this.props.toggleOption({ option, clearOthers, forceSelect: false });
-    this.onHideOptions();
-  };
+    onToggleOption = (option: VariableOption, clearOthers: boolean) => {
+      const toggleFunc = this.hasMultiPropertyAssigned(this.props.variable)
+        ? this.onToggleMultiValueVariable
+        : this.onToggleSingleValueVariable;
+      toggleFunc(option, clearOthers);
+    };
 
-  onToggleMultiValueVariable = (option: VariableOption, clearOthers: boolean) => {
-    this.props.toggleOption({ option, clearOthers, forceSelect: false });
-  };
+    onToggleSingleValueVariable = (option: VariableOption, clearOthers: boolean) => {
+      this.props.toggleOption({ option, clearOthers, forceSelect: false });
+      this.onHideOptions();
+    };
 
-  render() {
-    const { variable, picker } = this.props;
-    const showOptions = picker.id === variable.id;
+    onToggleMultiValueVariable = (option: VariableOption, clearOthers: boolean) => {
+      this.props.toggleOption({ option, clearOthers, forceSelect: false });
+    };
 
-    return (
-      <div className="variable-link-wrapper">
-        {this.renderLink(showOptions, variable)}
-        {this.renderOptions(showOptions, picker)}
-      </div>
-    );
-  }
+    hasMultiPropertyAssigned = (variable: any) => {
+      return variable.hasOwnProperty('multi') && variable.multi === true;
+    };
 
-  renderLink(showOptions: boolean, variable: VariableWithMultiSupport) {
-    if (showOptions) {
-      return null;
-    }
+    render() {
+      const { variable, picker } = this.props;
+      const showOptions = picker.id === variable.id;
 
-    const linkText = formatVariableLabel(variable);
-    const tags = getSelectedTags(variable);
-    const loading = variable.state === LoadingState.Loading;
-
-    return (
-      <VariableLink
-        text={linkText}
-        tags={tags}
-        onClick={this.onShowOptions}
-        loading={loading}
-        onCancel={this.onCancel}
-      />
-    );
-  }
-
-  onCancel = () => {
-    getVariableQueryRunner().cancelRequest(toVariableIdentifier(this.props.variable));
-  };
-
-  renderOptions(showOptions: boolean, picker: OptionsPickerState) {
-    if (!showOptions) {
-      return null;
+      return (
+        <div className="variable-link-wrapper">
+          {this.renderLink(showOptions, variable)}
+          {this.renderOptions(showOptions, picker)}
```

Comment:
```suggestion
          {showOptions ? this.renderLink(variable) : this.renderOptions(picker)}
```
Maybe we could also simplify the logic a bit by moving `showOptions` out of the render functions. This would also indicate which component is rendered based on condition, without having to look inside the render functions.

Reply:
Yep, this was something that was already on my mind. I'll fix it!

Reply:
Fixed!